iot_packet_verifier/src/verifier.rs

```diff
@@ -180,74 +182,160 @@ pub trait ConfigServer {
 // consistent with BalanceCache
 pub struct CachedOrgClient {
     pub keypair: Keypair,
-    pub enabled_clients: HashMap<u64, bool>,
+    pub disabled_clients: HashSet<u64>,
     pub client: OrgClient<Channel>,
 }
 
 impl CachedOrgClient {
-    pub fn new(client: OrgClient<Channel>, keypair: Keypair) -> Self {
-        CachedOrgClient {
+    pub async fn new(
+        mut client: OrgClient<Channel>,
+        keypair: Keypair,
+    ) -> Result<Arc<Mutex<Self>>, tonic::Status> {
+        // Fetch all disables orgs:
+        let mut disabled_clients = HashSet::new();
+        let orgs = client.list(OrgListReqV1 {}).await?.into_inner();
+        for org in orgs.orgs {
+            if org.locked {
+                disabled_clients.insert(org.oui);
+            }
+        }
+        Ok(Arc::new(Mutex::new(CachedOrgClient {
             keypair,
-            enabled_clients: HashMap::new(),
+            disabled_clients,
             client,
+        })))
+    }
+
+    async fn enable_org(&mut self, oui: u64) -> Result<(), OrgClientError> {
+        tracing::info!(%oui, "enabling org");
+
+        self.disabled_clients.remove(&oui);
+        let mut req = OrgEnableReqV1 {
+            oui,
+            timestamp: Utc::now().timestamp_millis() as u64,
+            signer: self.keypair.public_key().into(),
+            signature: vec![],
+        };
+        let signature = self.keypair.sign(&req.encode_to_vec())?;
+        req.signature = signature;
+        let _ = self.client.enable(req).await?;
```

Comment:
we should make a note to have the responses validated at some point but not necessary for this PR

Reply:
definitely